src/main.rs

```diff
@@ -169,15 +170,40 @@ where
     // allow interior mutibility (see LndNodeSigner) so this function can't safely be passed off to another thread.
     // This function is expected to finish if any producing thread exits (because we're not longer receiving the
     // events we need).
-    let consume_result = consume_messenger_events(&onion_messenger, receiver).await;
-    match consume_result {
-        Ok(_) => info!("Consume messenger events exited"),
-        Err(e) => error!("Consume messenger events exited: {e}"),
+    let consume_fut = consume_messenger_events(&onion_messenger, receiver);
+
+    // Similar to consume_messenger_events, relay_messenger_events needs to be run asyncronously, but can't be safely
+    // passed to another thread.
```

Comment:
Hm this is a bit of a nightnare (curses, `RefCell`). What do you think about trying to handle this inside of `consume_messenger_events` so that we don't need to deal with these unmovable tasks: 

Would look something like: 
1. Very simple producer loop that just ticks and passes a `MessengerEvents::Relay`
2. `consume_messenger_events` then calls `relay_messenger_events` to drain the queue. 

Nice to have all the onion messenger events in one spot / single select (helps simplofy the mental model / fewer races etc).

Reply:
Ah yes I like the idea of having all the onion messenger events in one spot. So to make sure I'm understanding: With this we'll be splitting relay_messenger_events into two functions. We'd have a producer, called in this spot, that polls for any new messages. Then we'd consume/handle any new messages in consume_messenger_events?

Reply:
> We'd have a producer, called in this spot, that polls for any new messages. Then, after passing these new messages into a channel, we'd consume/handle any new messages in consume_messenger_events?

The producer would just accept a ticker and send a `MessengerEvents::SendOutgoing` event at regular intervals (it doesn't have access to the onion messenger, so wouldn't have the actual message). We can then handle that message type in `consume_messenger_events` as follows: 
1. Check whether `OnionMessenger` has any pending messages
2. Send them to LND via `send_custom_message`

Reply:
Ah got it, yup that makes sense to me